[CI] MultiClusterSpecIT test {string.NotEqualsNullToUpperFolded} failing

**Build Scans:**
- [elasticsearch-periodic #5557 / 8.13.4_bwc](https://gradle-enterprise.elastic.co/s/tuxxh64lyfsku)
- [elasticsearch-periodic #5552 / 8.13.4_bwc](https://gradle-enterprise.elastic.co/s/hljtrrydsmhai)
- [elasticsearch-periodic #5547 / 8.13.4_bwc](https://gradle-enterprise.elastic.co/s/gsmsnyfwdrxao)
- [elasticsearch-periodic #5542 / 8.13.4_bwc](https://gradle-enterprise.elastic.co/s/cthrbqvkwuzww)
- [elasticsearch-periodic #5537 / 8.13.4_bwc](https://gradle-enterprise.elastic.co/s/s3ta4t4xpv7xk)
- [elasticsearch-periodic #5532 / 8.13.4_bwc](https://gradle-enterprise.elastic.co/s/47ttwk23hqfyi)
- [elasticsearch-periodic #5527 / 8.13.4_bwc](https://gradle-enterprise.elastic.co/s/rbkq2mcwth5qy)
- [elasticsearch-periodic #5522 / 8.13.4_bwc](https://gradle-enterprise.elastic.co/s/rfrweywmsm252)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:multi-clusters:v8.13.4#oldToNew" -Dtests.class="org.elasticsearch.xpack.esql.ccq.MultiClusterSpecIT" -Dtests.method="test {string.EqualsNullToUpperFolded}" -Dtests.seed=E8FEB11502A6A7B -Dtests.bwc=true -Dtests.locale=ccp-IN -Dtests.timezone=America/Rio_Branco -Druntime.java=23
```

**Applicable branches:**
8.x

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.ccq.MultiClusterSpecIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(test%20%7Bstring.NotEqualsNullToUpperFolded%7D),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
org.elasticsearch.client.ResponseException: method [POST], host [http://[::1]:33543], URI [/_query?pretty=true&error_trace=true], status line [HTTP/1.1 400 Bad Request]
¿eerror¿jroot_cause¿dtypeqparsing_exceptionfreasonx*line 2:37: token recognition error at: ':'kstack_tracey
<org.elasticsearch.xpack.esql.parser.ParsingException: line 2:37: token recognition error at: ':'
	at org.elasticsearch.xpack.esql.parser.EsqlParser$1.syntaxError(EsqlParser.java:109)
	at org.antlr.v4.runtime.ProxyErrorListener.syntaxError(ProxyErrorListener.java:41)
	at org.antlr.v4.runtime.Lexer.notifyListeners(Lexer.java:364)
	at org.antlr.v4.runtime.Lexer.nextToken(Lexer.java:144)
	at org.elasticsearch.xpack.esql.parser.EsqlParser$ParametrizedTokenSource.nextToken(EsqlParser.java:135)
	at org.antlr.v4.runtime.BufferedTokenStream.fetch(BufferedTokenStream.java:169)
	at org.antlr.v4.runtime.BufferedTokenStream.sync(BufferedTokenStream.java:152)
	at org.antlr.v4.runtime.BufferedTokenStream.consume(BufferedTokenStream.java:136)
	at org.antlr.v4.runtime.atn.ParserATNSimulator.execATN(ParserATNSimulator.java:515)
	at org.antlr.v4.runtime.atn.ParserATNSimulator.adaptivePredict(ParserATNSimulator.java:371)
	at org.elasticsearch.xpack.esql.parser.EsqlBaseParser.booleanExpression(EsqlBaseParser.java:826)
	at org.elasticsearch.xpack.esql.parser.EsqlBaseParser.whereCommand(EsqlBaseParser.java:630)
	at org.elasticsearch.xpack.esql.parser.EsqlBaseParser.processingCommand(EsqlBaseParser.java:535)
	at org.elasticsearch.xpack.esql.parser.EsqlBaseParser.query(EsqlBaseParser.java:316)
	at org.elasticsearch.xpack.esql.parser.EsqlBaseParser.singleStatement(EsqlBaseParser.java:199)
	at org.elasticsearch.xpack.esql.parser.EsqlParser.invokeParser(EsqlParser.java:73)
	at org.elasticsearch.xpack.esql.parser.EsqlParser.createStatement(EsqlParser.java:45)
	at org.elasticsearch.xpack.esql.session.EsqlSession.parse(EsqlSession.java:136)
	at org.elasticsearch.xpack.esql.session.EsqlSession.execute(EsqlSession.java:118)
	at org.elasticsearch.xpack.esql.execution.PlanExecutor.esql(PlanExecutor.java:67)
	at org.elasticsearch.xpack.esql.plugin.TransportEsqlQueryAction.innerExecute(TransportEsqlQueryAction.java:160)
	at org.elasticsearch.xpack.esql.plugin.TransportEsqlQueryAction.doExecuteForked(TransportEsqlQueryAction.java:137)
	at org.elasticsearch.xpack.esql.plugin.TransportEsqlQueryAction.lambda$doExecute$0(TransportEsqlQueryAction.java:121)
	at org.elasticsearch.server@8.13.4/org.elasticsearch.action.ActionRunnable$4.doRun(ActionRunnable.java:95)
	at org.elasticsearch.server@8.13.4/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
	at org.elasticsearch.server@8.13.4/org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:33)
	at org.elasticsearch.server@8.13.4/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:984)
	at org.elasticsearch.server@8.13.4/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: LexerNoViableAltException(':')
	at org.antlr.v4.runtime.atn.LexerATNSimulator.failOrAccept(LexerATNSimulator.java:309)
	at org.antlr.v4.runtime.atn.LexerATNSimulator.execATN(LexerATNSimulator.java:230)
	at org.antlr.v4.runtime.atn.LexerATNSimulator.match(LexerATNSimulator.java:114)
	at org.antlr.v4.runtime.Lexer.nextToken(Lexer.java:141)
	... 27 more
ÿÿdtypeqparsing_exceptionfreasonx*line 2:37: token recognition error at: ':'icaused_by¿dtypexlexer_no_viable_alt_exceptionfreasonökstack_tracey¡LexerNoViableAltException(':')
	at org.antlr.v4.runtime.atn.LexerATNSimulator.failOrAccept(LexerATNSimulator.java:309)
	at org.antlr.v4.runtime.atn.LexerATNSimulator.execATN(LexerATNSimulator.java:230)
	at org.antlr.v4.runtime.atn.LexerATNSimulator.match(LexerATNSimulator.java:114)
	at org.antlr.v4.runtime.Lexer.nextToken(Lexer.java:141)
	at org.elasticsearch.xpack.esql.parser.EsqlParser$ParametrizedTokenSource.nextToken(EsqlParser.java:135)
	at org.antlr.v4.runtime.BufferedTokenStream.fetch(BufferedTokenStream.java:169)
	at org.antlr.v4.runtime.BufferedTokenStream.sync(BufferedTokenStream.java:152)
	at org.antlr.v4.runtime.BufferedTokenStream.consume(BufferedTokenStream.java:136)
	at org.antlr.v4.runtime.atn.ParserATNSimulator.execATN(ParserATNSimulator.java:515)
	at org.antlr.v4.runtime.atn.ParserATNSimulator.adaptivePredict(ParserATNSimulator.java:371)
	at org.elasticsearch.xpack.esql.parser.EsqlBaseParser.booleanExpression(EsqlBaseParser.java:826)
	at org.elasticsearch.xpack.esql.parser.EsqlBaseParser.whereCommand(EsqlBaseParser.java:630)
	at org.elasticsearch.xpack.esql.parser.EsqlBaseParser.processingCommand(EsqlBaseParser.java:535)
	at org.elasticsearch.xpack.esql.parser.EsqlBaseParser.query(EsqlBaseParser.java:316)
	at org.elasticsearch.xpack.esql.parser.EsqlBaseParser.singleStatement(EsqlBaseParser.java:199)
	at org.elasticsearch.xpack.esql.parser.EsqlParser.invokeParser(EsqlParser.java:73)
	at org.elasticsearch.xpack.esql.parser.EsqlParser.createStatement(EsqlParser.java:45)
	at org.elasticsearch.xpack.esql.session.EsqlSession.parse(EsqlSession.java:136)
	at org.elasticsearch.xpack.esql.session.EsqlSession.execute(EsqlSession.java:118)
	at org.elasticsearch.xpack.esql.execution.PlanExecutor.esql(PlanExecutor.java:67)
	at org.elasticsearch.xpack.esql.plugin.TransportEsqlQueryAction.innerExecute(TransportEsqlQueryAction.java:160)
	at org.elasticsearch.xpack.esql.plugin.TransportEsqlQueryAction.doExecuteForked(TransportEsqlQueryAction.java:137)
	at org.elasticsearch.xpack.esql.plugin.TransportEsqlQueryAction.lambda$doExecute$0(TransportEsqlQueryAction.java:121)
	at org.elasticsearch.server@8.13.4/org.elasticsearch.action.ActionRunnable$4.doRun(ActionRunnable.java:95)
	at org.elasticsearch.server@8.13.4/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
	at org.elasticsearch.server@8.13.4/org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:33)
	at org.elasticsearch.server@8.13.4/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:984)
	at org.elasticsearch.server@8.13.4/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
ÿkstack_tracey
<org.elasticsearch.xpack.esql.parser.ParsingException: line 2:37: token recognition error at: ':'
	at org.elasticsearch.xpack.esql.parser.EsqlParser$1.syntaxError(EsqlParser.java:109)
	at org.antlr.v4.runtime.ProxyErrorListener.syntaxError(ProxyErrorListener.java:41)
	at org.antlr.v4.runtime.Lexer.notifyListeners(Lexer.java:364)
	at org.antlr.v4.runtime.Lexer.nextToken(Lexer.java:144)
	at org.elasticsearch.xpack.esql.parser.EsqlParser$ParametrizedTokenSource.nextToken(EsqlParser.java:135)
	at org.antlr.v4.runtime.BufferedTokenStream.fetch(BufferedTokenStream.java:169)
	at org.antlr.v4.runtime.BufferedTokenStream.sync(BufferedTokenStream.java:152)
	at org.antlr.v4.runtime.BufferedTokenStream.consume(BufferedTokenStream.java:136)
	at org.antlr.v4.runtime.atn.ParserATNSimulator.execATN(ParserATNSimulator.java:515)
	at org.antlr.v4.runtime.atn.ParserATNSimulator.adaptivePredict(ParserATNSimulator.java:371)
	at org.elasticsearch.xpack.esql.parser.EsqlBaseParser.booleanExpression(EsqlBaseParser.java:826)
	at org.elasticsearch.xpack.esql.parser.EsqlBaseParser.whereCommand(EsqlBaseParser.java:630)
	at org.elasticsearch.xpack.esql.parser.EsqlBaseParser.processingCommand(EsqlBaseParser.java:535)
	at org.elasticsearch.xpack.esql.parser.EsqlBaseParser.query(EsqlBaseParser.java:316)
	at org.elasticsearch.xpack.esql.parser.EsqlBaseParser.singleStatement(EsqlBaseParser.java:199)
	at org.elasticsearch.xpack.esql.parser.EsqlParser.invokeParser(EsqlParser.java:73)
	at org.elasticsearch.xpack.esql.parser.EsqlParser.createStatement(EsqlParser.java:45)
	at org.elasticsearch.xpack.esql.session.EsqlSession.parse(EsqlSession.java:136)
	at org.elasticsearch.xpack.esql.session.EsqlSession.execute(EsqlSession.java:118)
	at org.elasticsearch.xpack.esql.execution.PlanExecutor.esql(PlanExecutor.java:67)
	at org.elasticsearch.xpack.esql.plugin.TransportEsqlQueryAction.innerExecute(TransportEsqlQueryAction.java:160)
	at org.elasticsearch.xpack.esql.plugin.TransportEsqlQueryAction.doExecuteForked(TransportEsqlQueryAction.java:137)
	at org.elasticsearch.xpack.esql.plugin.TransportEsqlQueryAction.lambda$doExecute$0(TransportEsqlQueryAction.java:121)
	at org.elasticsearch.server@8.13.4/org.elasticsearch.action.ActionRunnable$4.doRun(ActionRunnable.java:95)
	at org.elasticsearch.server@8.13.4/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
	at org.elasticsearch.server@8.13.4/org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:33)
	at org.elasticsearch.server@8.13.4/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:984)
	at org.elasticsearch.server@8.13.4/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: LexerNoViableAltException(':')
	at org.antlr.v4.runtime.atn.LexerATNSimulator.failOrAccept(LexerATNSimulator.java:309)
	at org.antlr.v4.runtime.atn.LexerATNSimulator.execATN(LexerATNSimulator.java:230)
	at org.antlr.v4.runtime.atn.LexerATNSimulator.match(LexerATNSimulator.java:114)
	at org.antlr.v4.runtime.Lexer.nextToken(Lexer.java:141)
	... 27 more
ÿfstatusÿ
```

**Issue Reasons:**
- [8.x] 8 failures in test test {string.NotEqualsNullToUpperFolded} (1.3% fail rate in 607 executions)
- [8.x] 8 failures in step 8.13.4_bwc (100.0% fail rate in 8 executions)
- [8.x] 8 failures in pipeline elasticsearch-periodic (100.0% fail rate in 8 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/x-pack/plugin/esql/qa/testFixtures/src/main/resources/string.csv-spec
+++ b/x-pack/plugin/esql/qa/testFixtures/src/main/resources/string.csv-spec
@@ -1317,7 +1317,7 @@ emp_no:integer  | first_name:keyword
 
 equalsNullToUpperFolded
 from employees
-| where to_upper(first_name) == null::keyword
+| where to_upper(first_name) == to_keyword(null)
 | keep emp_no, first_name
 ;
 

--- a/x-pack/plugin/esql/qa/testFixtures/src/main/resources/string.csv-spec
+++ b/x-pack/plugin/esql/qa/testFixtures/src/main/resources/string.csv-spec
@@ -1317,7 +1317,7 @@ emp_no:integer  | first_name:keyword
 
 equalsNullToUpperFolded
 from employees
-| where to_upper(first_name) == to_keyword(null)
+| where to_upper(first_name) == to_string(null)
 | keep emp_no, first_name
 ;
 
@@ -1335,7 +1335,7 @@ emp_no:integer  | first_name:keyword
 
 notEqualsNullToUpperFolded
 from employees
-| where to_upper(first_name) != null::keyword
+| where to_upper(first_name) != to_string(null)
 | keep emp_no, first_name
 ;
 

